File: api-scala/src/test/scala/com/ultreon/craft/scala/testmod/init/ModItems.scala
package dev.ultreon.quantum.scala.testmod.init

import dev.ultreon.quantum.item.Item
import dev.ultreon.quantum.registry.{DeferRegistry, Registries, DeferredElement}
import dev.ultreon.quantum.scala.testmod.Constants

object ModItems {
  private final val REGISTER = DeferRegistry.of(Constants.MOD_ID, Registries.ITEM)

  final val TEST_ITEM: DeferredElement[Item] = REGISTER.defer("test_item", { () =>
    new Item(new Item.Properties)
  })

  def register(): Unit = {
    REGISTER.register()
  }
}

File: testmod/src/main/java/io/github/xypercode/init/ModItems.java
package io.github.xypercode.init;

import dev.ultreon.quantum.item.Item;
import dev.ultreon.quantum.registry.DeferRegistry;
import dev.ultreon.quantum.registry.DeferredElement;
import dev.ultreon.quantum.registry.Registries;
import io.github.xypercode.TestMod;

public class ModItems {
    private static final DeferRegistry<Item> REGISTER = DeferRegistry.of(TestMod.MOD_ID, Registries.ITEM);

    public static final DeferredElement<Item> TEST_ITEM = REGISTER.defer("test_item", () -> new Item(new Item.Properties()));

    public static void register() {
        REGISTER.register();
    }
}
